packages/cpp/src/browser/cpp-build-configurations-ui.ts

```diff
@@ -67,15 +82,20 @@ export class CppBuildConfigurationChanger implements QuickOpenModel {
             if (mode !== QuickOpenMode.OPEN) {
                 return false;
             }
-            this.commandService.executeCommand(CPP_RESET_BUILD_CONFIGURATION.id);
+            this.cppBuildConfigurations.setActiveConfig(undefined, this.selectedRoot);
             return true;
         },
     });
 
     async onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
```

Comment:
Can we replace this by something more lightweight, such as `quickPick`?
(https://github.com/theia-ide/theia/blob/master/packages/editor/src/browser/editor-command.ts#L159)

It would remove the state that you keep too.
(https://github.com/theia-ide/theia/pull/4603/files#diff-675a6436a80a79d155950aa7d931bef6R62)

Reply:
sure, I'll try and refactor and simplify as much as possible.